backend/src/main/java/com/htecgroup/skynest/service/impl/PermissionServiceImpl.java

```diff
@@ -237,6 +237,21 @@ public void revokePermission(UUID bucketId, UUID userId) {
     permissionRepository.delete(permission);
   }
 
+  @Override
+  public void revokeFilePermission(UUID fileId, UUID userId) throws FileNotFoundException {
+    FileMetadataEntity fileMetadataEntity =
+        fileRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+    checkIfBucketIsDeleted(fileMetadataEntity.getBucket().getId());
+    currentUserHasPermissionForFile(fileMetadataEntity, AccessType.OWNER);
+    currentUserHasPermissionForFolder(fileMetadataEntity.getParentFolder(), AccessType.OWNER);
+    currentUserHasPermissionForBucket(fileMetadataEntity.getBucket().getId(), AccessType.OWNER);
+
+    UserEntity user = userRepository.findById(userId).orElseThrow(UserNotFoundException::new);
+    UserObjectAccessEntity permission =
+        permissionRepository.findByObjectIdAndGrantedTo(fileId, user);
```

Comment:
this doesn't return an optional? should we refactor?
if this returns a null, we should throw an error?

Reply:
It does return Optional, but it was asking  for throw in method signature as well, I can't say why.